### ISSUE TITLE
Can't paste into TextExpander

### ISSUE THREAD

**Issue by slikk66** [TermiT/Flycut#102]
I LOVE flycut, and I've recently got TextExpander which is another super fancy tool to keep snippets of code to expand into commands.

i.e. type   ;icert 

And out pops:
aws iam upload-server-certificate --server-certificate-name example.com-v1 --certificate-body file://example.com.crt --private-key file://example.com.key --path=/cloudfront/

However, with flycut, you can't paste into the boxes of text expander.

I don't know which app is at fault, but when you're in the terminal and paste with flycut, even though my mouse cursor is in the text expander boxes, pasting pastes to terminal.

This happens it seems with any window where the cursor is, then I do my Text Expander abbreviation, the TE window pops up, then I initiate FlyCut and try to paste into the TE window, it instead appears behind where the cursor originally was.  Is there any way to fix this?

**Comment by slikk66** [comment]
Received a response from Smile Support (makers of text expander) they were able to reproduce the problem but blamed the actions on FlyCut, not on their software.. FYI

**Comment by MarkJerde** [comment]
Hi slikk66,
It sounds like TextExpander listens to what you are typing ("type ;icert") and inserts text in response to it?  I've glanced at the videos on Smile's website and can't figure out what "the boxes of text expander" are.  It sounds like something other than a normal window.  Maybe a video or picture would help.  If they are using non-normal windows it doesn't surprise me much that Flycut would have trouble with them, due to the tricks Flycut uses to activate and then return to what was previously active and paste.

Have you tried pasting from both the bezel and the menu?  I can't recall how the menu works, but I'd be curious if that makes a difference.

Thanks,
Mark

**Comment by slikk66** [comment]
Hi Mark,

Thanks for replying.  I've attached an image here of my normal workflow with both FlyCut and TextExpander.  I got a reply from their support also, I'll put that in here as well.
<img width="1171" alt="screen shot 2016-03-30 at 7 53 34 pm" src="https://cloud.githubusercontent.com/assets/633269/14164100/3a0bd4d8-f6b1-11e5-9789-95e89490b51c.png">

The image shows me in terminal full screen, running a Text Expander snippet (by typing ;icert) which then gets removed, and the snippet input window appears, then loading FlyCut (you can see the overlay menu) and trying to paste into the box in the window, you can see the highlighted box in the top right of the floating window, when attempting to paste into it, the IP I have selected gets pasted into the terminal prompt rather than the highlighted box.  I opened up the FlyCut window again just for reference.

Since I'm in terminal full screen, I don't have the window ribbon menu, but I did test it out of fullscreen mode and the first paste didn't work selecting it from the drop down, same issue, but the next ones did.. I was able to paste in a few different values.  So if I'm in multi screen mode, that may be a work around, thanks for the suggestion!

However, the full screen + FlyCut pop up window issue remains.

Here's the response from their support:

```
Hi Dan,

Thanks for contacting Smile Support. I'm sorry to hear about the difficulty. I'll be happy to answer your questions.

I downloaded FlyCut to test it out and see if I ran into the same issue. I can see the problem. It looks like FlyCut isn't detecting the proper window that has focus (the TextExpander popup window for your snippet) and so FlyCut pastes the clipboard selection that you choose into the parent window (e.g., text editor) from which you invoked the snippet.

Unfortunately, I don't think this is under TextExpander's control. Many other apps work correctly with TextExpander popups, and TextExpander is fully compatible with Mac OS X 10.11 El Capitan. This seems to be a bug in FlyCut, where it's not reading the focused window correctly. From my quick test so far, FlyCut looks a very nice, functional piece of software, but it was last updated in December 2011. So I think it's quite likely the bug is in their app, not TextExpander.

I'm sorry not have a better solution to help you with this. You may want to try to contact the developer who maintains FlyCut. I see in its Preferences > Acknowledgements that it was developed by General Arcade (http://www.generalarcade.com/)... also on the Mac App Store the name Gennadiy Potapov is mentioned. I'm not sure if it's still being actively developed, but to address this issue, contacting them would be the route to take.

Please let us know if you have any other comments, questions, or concerns. We're always happy to help.

Thanks for using TextExpander from Smile!
```

**Comment by Juberstine** [comment]
I have this same issue as well and would love to see it fixed. Another clipboard manager, LaunchBar, does not suffer from the same issue, so clearly this can be fixed :-)

**Comment by JoelStickney** [comment]
Same issue here, FlyCut needs to properly support "background apps" (NoteFile, Fantastical, etc) which it currently does not.

**Comment by MarkJerde** [comment]
@slikk66, @Juberstine, and @JoelStickney - I just put in a fix for #45 that was suggested by another contributor which seems like a possible candidate for fixing this issue as well.

If any of you could try the beta to see if it fixes your problem I would really appreciated it.  You can download it from [https://github.com/MarkJerde/LMBTFY/blob/master/Flycut-Issue45-Beta.tgz?raw=true](https://github.com/MarkJerde/LMBTFY/blob/master/Flycut-Issue45-Beta.tgz?raw=true).  This build uses the Flycut preferences/save in ~/Library/Preferences/com.generalarcade.flycut.plist (this moves around depending on code-signing settings, so documented here so you don't have trouble losing your clippings list).

Thanks,
Mark

**Comment by slikk66** [comment]
Hi Mark,

Finally downloaded the new version from releases - this problem has been fixed.

Thanks!
